app/cp_nigeria/views.py

```diff
@@ -825,9 +826,12 @@ def cpn_constraints(request, proj_id, step_id=STEP_MAPPING["economic_params"]):
             form_errors = True
 
         if equity_form.is_valid():
-            equity_data = equity_form.save(commit=False)
-            equity_data.debt_start = scenario.start_date.year
-            equity_data.scenario = scenario
+            if equity_data is not None:
```

Comment:
I am not sure why this is needed, normally `.save` should update existing fields' values

Reply:
Yes, I am not sure what is happening here either. It seems like since creating the custom clean method, when the form is validated it is no longer bound to the instance even though it is initialized with it (I checked the form with a pdb trace and the instance was None), so I did this quick fix so that a new model wouldn't be created causing the error you described. Could not figure out why it was behaving like this either.